packages/encryptor/src/index.ts

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { decrypt as metamaskDecrypt, encrypt, EncryptionResult } from '@metamask/browser-passworder'
+import { Buffer } from 'buffer'
+
+// Polyfill Buffer if we are in a browser environment
+if (typeof window !== 'undefined') window.Buffer = Buffer
```

Comment:
Q: Why do that here in the encryptor package?

Reply:
That's a good question!

So, I tried using `@metamask/browser-passworder` directly on the desktop wallet. However, I got the error that `Buffer is undefined` (so much for "browser"). This means it needs polyfilling to be used on the desktop wallet. Potentially that'd also be the case for mobile wallet, or any other app we might wanna use it. Because of this (and because of the need for the small utilities that I wrote) I decided to go for our own package and polyfill it only once.

To more specifically answer your question: The above line detects if we are in a browser environment (like React for example). In that case, it polyfills Buffer with the one coming from the NPM package `buffer`. Otherwise (if we are running the test suite for example in a node environment) it won't polyfill it.